lib/Sema/CSDiagnostics.cpp

```diff
@@ -3399,6 +3402,38 @@ bool MissingMemberFailure::diagnoseForDynamicCallable() const {
   return false;
 }
 
+bool MissingMemberFailure::diagnoseForDefaultAnyInCollection() const {
+  auto &cs = getConstraintSystem();
+  auto *expr = getAsExpr(getAnchor());
+  auto *parentExpr = cs.getParentExpr(expr);
+  auto &solution = getSolution();
+
+  if (parentExpr && isa<UnresolvedMemberExpr>(expr)) {
```

Comment:
Could you please turn it into a shortcut like `if (!(parentExpr && isa<UnresolvedMemberExpr>(expr))) return false;`?